rest/main.go

```diff
@@ -146,3 +158,144 @@ func serverMainPersistentConfig(fs *flag.FlagSet, flagStartupConfig *StartupConf
 
 	return false, startServer(&sc, ctx)
 }
+
+func automaticConfigUpgrade(configPath string) (*StartupConfig, bool, error) {
+	legacyServerConfig, err := LoadServerConfig(configPath)
+	if err != nil {
+		return nil, false, err
+	}
+
+	if legacyServerConfig.DisablePersistentConfig != nil && *legacyServerConfig.DisablePersistentConfig {
+		return nil, true, nil
+	}
+
+	startupConfig, dbConfigs, err := legacyServerConfig.ToStartupConfig()
+	if err != nil {
+		return nil, false, err
+	}
+
+	dbConfigs, err = sanitizeDbConfigs(dbConfigs)
+	if err != nil {
+		return nil, false, err
+	}
+
+	// Attempt to establish connection to server, add retry like its other use-case
+	cluster, err := EstablishCouchbaseClusterConnection(startupConfig)
+	if err != nil {
+		return nil, false, err
+	}
+
+	defer func() {
+		_ = cluster.Close()
+	}()
+
+	// Write database configs to CBS with groupID "default"
+	for _, dbConfig := range dbConfigs {
+		_, err = cluster.PutConfig(*dbConfig.Bucket, "default", base.Uint64Ptr(0), dbConfig)
```

Comment:
There's already a const for default group ID

```suggestion
		_, err = cluster.PutConfig(*dbConfig.Bucket, persistentConfigDefaultGroupID, base.Uint64Ptr(0), dbConfig)
```